src/in_progress.rs

```diff
@@ -1110,3 +1110,132 @@ pub fn is_hardware_waiting_repair(
     let stmt_query = conn.query(&stmt, &params)?;
     Ok(!stmt_query.is_empty())
 }
+
+/// Get region id based on the region name.
+pub fn get_region_id(pool: &Pool<ConnectionManager>, region_name: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get region Id from region name
+    let stmt = "SELECT region_id FROM regions WHERE region_name = $1";
+    let stmt_query = conn.query(stmt, &[&region_name])?;
+
+    let mut region_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return region_id
+        let id: i32 = res.get(0);
+        region_id = id as u32;
+        debug!("Region id {} for the region {}", region_id, region_name);
+        Ok(Some(region_id))
+    } else {
+        // does not exist
+        debug!("No region with name {} in database", region_name);
+        Ok(None)
+    }
+    
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_id(pool: &Pool<ConnectionManager>, storage_type: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage Id from storage type
+    let stmt = "SELECT storage_id FROM storage_types WHERE storage_type= $1 ";
+    let stmt_query = conn.query(&stmt, &[&storage_type])?;
+
+    let mut storage_id: u32 = 0;
```

Comment:
Same here.  This variable isn't needed anymore :)

Reply:
Remove variable